runtime/lua/vim/fs.lua

```diff
@@ -350,6 +350,9 @@ function M.normalize(path, opts)
 
   path = path:gsub('\\', '/'):gsub('/+', '/')
 
+  if path == '/' then
+    return path
+  end
```

Comment:
Since this comes after the `gsub` on line 351, won't this mean that `vim.fs.normalize("\\")` will return `"/"`? I'm not sure if that's correct.

I am actually not sure _what_ `vim.fs.normalize("\\")` should return. It's an invalid path. It should probably either return `""` (empty string to indicate invalid path) or `"\\"` (original value unmodified) to indicate that it's invalid. But I'm pretty sure it should not return `"/"`.

We don't have to solve this particular issue in this PR, but I do think this should check should come earlier in the function (probably right after `vim.validate`).

```lua
if path:match('^/+$') then
  return '/'
end
```

Reply:
(I may be missing some context so feel free to ignore)

`\\` at the start of of a path is valid in almost all programs and languages,
(including lua, libuv, and vimscript of course)
it gets expanded to `C:\\`, so I think that should be the same here